.github/workflows/docker.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Docker images
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/docker.yml'
+      - 'docker/**'
+  schedule:
+    - cron: '42 7 * * *' # run at 7:42 UTC (morning) every day
+jobs:
+  build:
+    name: Build and push
+    runs-on: ubuntu-latest
+    env:
+      RUST_IMAGE_TAG: latest
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          docker login --username $DOCKERHUB_USERNAME --password $DOCKERHUB_TOKEN
+
+          RUST_IMAGE=rust:$RUST_IMAGE_TAG
+          docker run --rm $RUST_IMAGE rustc --version
+          RUST_VERSION=$(docker run --rm $RUST_IMAGE rustc --version)
+          echo "Rust version in $RUST_IMAGE is $RUST_VERSION."
+
+          CHEF_IMAGE=$DOCKERHUB_USERNAME/cargo-chef:$RUST_IMAGE_TAG
+          CHEF_RUST_VERSION=$(docker run --rm $CHEF_IMAGE rustc --version || echo "none")
+          echo "Rust version in $CHEF_IMAGE is $CHEF_RUST_VERSION."
+          CHEF_VERSION=$(docker run --rm $CHEF_IMAGE cargo install --list | grep "^cargo-chef v" | cut -d v -f 2 | cut -d : -f 1 || echo "none")
```

Comment:
This is a bit cheesy, but `cargo chef --version` currently returns a dummy value.

Even if it did, the script would depend on the fact that it returns _exactly same_ value as `cargo search cargo-chef` - i.e. any fancy stuff like `1.49.0 (e1884a8e3 2020-12-29)` would break it.

Reply:
If this job would run on releases, i.e.
```
on:
  release:
    types:
      - published
```
then it would be possible to use `${{ github.event.release.tag_name }}` as a version tag for an image. (That also would be a fix for the dummy `--version` output).

Reply:
The job will have to run a schedule to release a new version every time Rust releases, even if cargo-chef did not cut a new release.
But @TriplEight makes a very good point - we can rely on the latest non-alpha tag in the repository to detect the latest version of `cargo-chef`, no need to touch Docker.

Reply:
For the Rust version - we could use a [label](https://docs.docker.com/engine/reference/builder/#label)

Reply:
Sorry, I didn't quite get the idea behind running this job on schedule for every new Rust. Can you elaborate, please? 
You do not even test the code against `nightly`. If it would be the case, and if you need to release a container for every push, just place this job right after one which would run `cargo +nightly test ...`.

Reply:
> The job will have to run a schedule to release a new version every time Rust releases, even if cargo-chef did not cut a new release.

Right, I was about to say exactly this.

> But @TriplEight makes a very good point - we can rely on the latest non-alpha tag in the repository to detect the latest version of cargo-chef, no need to touch Docker.

Well, this particular command here is to query the version of cargo chef _in the latest published docker image_ (not latest version published to crates.io).

Reply:
> Sorry, I didn't quite get the idea behind running this job on schedule for every new Rust. Can you elaborate, please?

We are referring to stable Rust releases.
E.g. if the latest cargo chef is `0.1.12` and we build the Docker image today, the released Docker image will be using Rust `1.49.0`. When Rust `1.50.0` is released, we want to release a new image with cargo-chef `0.1.12` and Rust `1.50.0`.

Reply:
> Sorry, I didn't quite get the idea behind running this job on schedule for every new Rust. Can you elaborate, please?

It is somewhat briefly described in #41 and its comments. The idea is to have have up-to-date images with latest released cargo-chef, eventually for multiple tags of the official `rust` image, but starting with just `latest` in this PR (but a matter of setting one variable). Notice that release cadence of `rust` (including minor versions) may be actually higher than of `cargo chef`, hence a need for a schedule.

> For the Rust version - we could use a label

Yes it would be technically possible to label $CHEF_IMAGE (though we have no power to label $RUST_IMAGE), and then read it (which would still AFAICS require pulling the image). But in order to label it, we would still need to determine the version (or rust in $RUST_IMAGE) somehow? Maybe that's just a lack of my imagination - feel free to suggest concrete changes to code.